spec.html

```diff
@@ -32997,7 +32997,7 @@ <h1>%ArrayIteratorPrototype%.next ( )</h1>
             1. If _itemKind_ is `"value"`, let _result_ be _elementValue_.
             1. Else,
               1. Assert: _itemKind_ is `"key+value"`.
-              1. Let _result_ be CreateArrayFromList(&laquo; _index_, _elementValue_ &raquo;).
+              1. Let _result_ be ! CreateArrayFromList(&laquo; _index_, _elementValue_ &raquo;).
```

Comment:
this is misleading for me. When I read the specs for tests I use the `!` as a hint for: "this abstraction could possibly return an abrupt completion, but won't return any for the given arguments". In this same case,  `CreateArrayFromList` won't ever return any abrupt completion, therefore, the `!` is unnecessary and requires me extra time trying to figure out where it could possibly return an abrupt completion until I realize it won't.

This happens every single time I read the specs.

I'm curious what would be the advantage of having the `!`. I believe the specs are way more consistent when they are meaningful.

Reply:
According to @domenic's [comment in `Promise.allSettled` repo](https://github.com/tc39/proposal-promise-allSettled/issues/32#issue-417895560), `!` or `?` should be used before every abstract operation except if its result is completion value.

Reply:
We should be using it every time; every abstract operation could possibly return an abrupt completion, when viewed from the call site.

Reply:
that's really unhelpful and I can't rediscuss everything again. I already had mentioned that in another issue but this is circling back.

I can't block anything, but assuming we know our abstractions, using a flag - `!` - only when it's necessary makes it more informative. The referred abstractions are not coming from unknown sources, they are specified in the same document.

Using `!` for everything only makes it vague and redundant. We can be more assertive than that, we already are, no need for changes.

Reply:
We should probably move that discussion to #486; at the moment, every single abstract operation call in the spec should either have its completion record return inspected, or should have a ! or a ? (there’s of plenty of places left to fix, but we should fix them whenever they come up)

Reply:
There is nothing else I can add to the discussion here or there. I can't mandate what individually want and I don't think I have more arguments (as in facts) to convince anyone to change their idea.

It is what it is. At least this one is only editorial.